src/runtime/HalidePyTorchHelpers.h

```diff
@@ -90,16 +91,29 @@ inline Buffer<scalar_t> wrap(at::Tensor &tensor) {
 #endif
     Buffer<scalar_t> buffer;
 
-    // TODO(mgharbi): force Halide to put input/output on GPU?
-    if (tensor.is_cuda()) {
-        buffer = Buffer<scalar_t>(dims);
-        const halide_device_interface_t *cuda_interface = halide_cuda_device_interface();
-        int err = buffer.device_wrap_native(cuda_interface, (uint64_t)pData);
-        AT_ASSERTM(err == 0, "(CUDA) halide_device_wrap failed");
-        buffer.set_device_dirty();
-    } else {
-        buffer = Buffer<scalar_t>(pData, dims);
-    }
+    buffer = Buffer<scalar_t>(pData, dims);
+
+    return buffer;
+}
+
+
+template<class scalar_t>
+inline Buffer<scalar_t> wrap_cuda(at::Tensor &tensor) {
+    check_type<scalar_t>(tensor);
+    std::vector<int> dims = get_dims(tensor);
+#if HL_PYTORCH_API_VERSION >= 13
+    scalar_t *pData = tensor.data_ptr<scalar_t>();
+#else
+    scalar_t *pData = tensor.data<scalar_t>();
+#endif
+    Buffer<scalar_t> buffer;
```

Comment:
...why not

    Buffer<scalar_t> buffer(dims);

and then delete line 112?

Reply:
done